marda_extractors_api/__init__.py

```diff
@@ -395,25 +402,43 @@ def apply_template_args(
         input_path: Path,
         output_type: str | None = None,
         output_path: Path | None = None,
-    ):
+        additional_template: dict | None = None,
+    ) -> str:
         """Reference implementation of templating in MaRDA Metadata Extractor Schemas.
 
         See the :doc:`mme_schema:mme_schema/UsageTemplate` for details of the individual arguments.
+
+        Parameters:
+            command: The command to apply the template to.
+            method: The execution method to use.
+            input_type: The input type to use.
+            input_path: The input path to use.
+            output_type: The output type to use.
+            output_path: The output path to use.
+            additional_template: Additional template arguments to use, which
+                overwrite the default arguments.
+
+        Returns:
+            The templated command.
+
         """
-        if method == SupportedExecutionMethod.CLI:
-            command = command.replace("{{ input_type }}", f"marda:{input_type}")
-            command = command.replace("{{ input_path }}", str(input_path))
-            if output_path:
-                command = command.replace("{{ output_path }}", str(output_path))
-            if output_type:
-                command = command.replace("{{ output_type }}", str(output_type))
-        else:
-            command = command.replace("{{ input_type }}", f"{str(input_type)!r}")
-            command = command.replace("{{ input_path }}", f"{str(input_path)!r}")
-            if output_path:
-                command = command.replace("{{ output_path }}", f"{str(output_path)!r}")
-            if output_type:
-                command = command.replace("{{ output_type }}", f"{str(output_type)!r}")
+        if additional_template is None:
+            additional_template = {}
+        default_fields = {"input_type", "input_path", "output_type", "output_path"}
+        for field in default_fields:
+            value = additional_template.get(field) or locals()[field]
+            if method == SupportedExecutionMethod.CLI:
+                command = command.replace(f"{{{{ {field} }}}}", str(value))
+            else:
+                command = command.replace(f"{{{{ {field} }}}}", f"{str(value)!r}")
+
+        for field in additional_template:
+            if field not in default_fields:
+                value = additional_template[field]
+                if method == SupportedExecutionMethod.CLI:
+                    command = command.replace(f"{{{{ {field} }}}}", str(value))
+                else:
+                    command = command.replace(f"{{{{ {field} }}}}", f"{str(value)!r}")
```

Comment:
Added the ability to put whatever variables you like in your template, not sure if this is currently allowed in the schema but might be useful for passing context